tests/phpunit/tests/blocks/getBlockTemplates.php

```diff
@@ -221,4 +221,51 @@ public function data_get_block_templates_should_respect_posttypes_property() {
 			),
 		);
 	}
+
+	/**
+	 * Test _get_block_templates_paths() for a valid theme dir.
+	 *
+	 * @ticket 58196
+	 * @covers ::_get_block_templates_paths
+	 */
+	public function test_get_block_templates_paths_dir_exists() {
+		$theme_dir = get_template_directory();
+		// templates in the current theme.
+		$templates = array(
+			'parts/small-header.html',
+			'templates/custom-single-post-template.html',
+			'templates/index.html',
+			'templates/page-home.html',
+			'templates/page.html',
+			'templates/single.html',
+		);
+
+		$expected_template_paths = array_map(
+			static function ( $template ) use ( $theme_dir ) {
+				return $theme_dir . '/' . $template;
+			},
+			$templates
+		);
+
+		$template_paths = _get_block_templates_paths( $theme_dir );
+		sort( $expected_template_paths );
+		sort( $template_paths );
+		$this->assertSame( $expected_template_paths, $template_paths );
+	}
+
+	/**
+	 * Test _get_block_templates_paths() for a invalid dir.
+	 *
+	 * @ticket 58196
+	 * @covers ::_get_block_templates_paths
+	 */
+	public function test_get_block_templates_paths_dir_doesnt_exists() {
+		$theme_dir               = '/tmp/random-invalid-theme-path';
+		$expected_template_paths = array();
+		// should return empty array for invalid path.
+		$template_paths = _get_block_templates_paths( $theme_dir );
+		sort( $expected_template_paths );
+		sort( $template_paths );
+		$this->assertSame( $expected_template_paths, $template_paths );
```

Comment:
This test can be simplified quite a bit since we're expecting an empty array.

```suggestion
		// Should return empty array for invalid path.
		$template_paths = _get_block_templates_paths( '/tmp/random-invalid-theme-path' );

		$this->assertEmpty( $template_paths );
```

Reply:
May be we can us `$this->assertSame( array(), $template_paths );` instead `assertEmpty`?

Reply:
I'd be happy with that. Probably a bit more specific.

Reply:
I have used `$this->assertSame( array(), $template_paths );`

I moved the test case to `tests/block-template.php` where some of the other util functions were.